source/fab/tasks/c.py

```diff
@@ -387,6 +389,30 @@ def run(self, artifacts: List[Artifact]) -> List[Artifact]:
                     state.add_c_dependency(current_def, node.spelling)
                     new_artifact.add_dependency(node.spelling)
 
+            elif node.kind == clang.cindex.CursorKind.VAR_DECL:
+                # Variable definitions can be external too, lodge any
+                # encountered in user headers here
+                logger.debug('  * Is a variable declaration')
+                if ((not node.is_definition())
+                        and node.linkage == clang.cindex.LinkageKind.EXTERNAL):
+                    if (self._check_for_include(node.location.line)
+                            == "usr_include"):
+                        logger.debug('  * Is defined elsewhere in the project')
+                        external_vars.append(node.spelling)
+
+            elif node.kind == clang.cindex.CursorKind.DECL_REF_EXPR:
+                # Find references to variables which came in externally (as
+                # captured by the list above)
+                logger.debug('  * Is a reference to a variable')
+                if node.spelling in external_vars and current_def is not None:
+                    # TODO: Assumption that the most recent exposed
+                    # definition encountered above is the one which
+                    # should lodge this dependency - is that true?
+                    logger.debug(
+                        '  * From elsewhere in the project (so a dependency)')
+                    state.add_c_dependency(current_def, node.spelling)
+                    new_artifact.add_dependency(node.spelling)
+
```

Comment:
I think it's pretty tied up with C because it's Clang's frontend doing the parsing, if we were doing this with Julia I guess we'd have a `tasks/julia.py` which has handling for Julia and then the various `tasks/*.py` would need extra logic to recognise whatever interfacing methods are used by Julia w/ those other languages

I do agree though that we ought to have something behind the language specific tasks that recognises that it's all just symbols - particularly because it would allow us to support e.g. identically named Fortran program units and C functions (which in reality is OK because the symbols are unique, but I think Fab presently would get confused by)